docs/topics/generics.md

```diff
@@ -322,6 +322,73 @@ fun <T> copyWhenGreater(list: List<T>, threshold: T): List<String>
 The passed type must satisfy all conditions of the `where` clause simultaneously. In the above example, the `T` type
 must implement _both_ `CharSequence` and `Comparable`.
 
+## Definitely non-nullable types
+
+To make interoperability with generic Java classes and interfaces easier, Kotlin supports declaring a generic type parameter
+as **definitely non-nullable**. 
+
+To declare a generic type `T` as definitely non-nullable, declare the type with `& Any`. For example: `T & Any`. 
+
+> The syntax represents an intersection between type `T` and the non-nullable type `Any`.
```

Comment:
I'm also not sure if this is needed. While this sentence is correct, Kotlin in general does not provide any kind of intersection types (besides this feature) and does not mention them anywhere in the documentation, pretending they are non-existent. Such `&` syntax is correct only for `& Any` case.

 Also, 'intersection type' is a kind of advanced topic in programming languages that may confuse inexperienced users.

Reply:
Removed!